lib/utils.js

```diff
@@ -96,7 +96,7 @@ var Utils = module.exports = {
   cloneDeep: function(obj, fn) {
     return lodash.cloneDeep(obj, function (elem) {
       // Preserve special data-types like `fn` across clones. _.get() is used for checking up the prototype chain
-      if (elem && typeof elem.clone === 'function') {return elem.clone(); }
+      if (elem && !Array.isArray(elem) && !lodash.isPlainObject(elem) && typeof elem.clone === 'function') {return elem.clone(); }
```

Comment:
Might read better to have seperate ifs for array and plain object and just return undefined